LayoutTests/http/tests/iframe-monitor/resources/monitor-setup.js

```diff
@@ -0,0 +1,50 @@
+async function setup() {
+    if (window.testRunner && 'canModifyResourceMonitorList' in testRunner && testRunner.canModifyResourceMonitorList) {
+        const rules = [
+            rule('--eligible--'),
+        ];
+
+        await testRunner.setResourceMonitorList(JSON.stringify(rules));
+
+        // Lower the threshold to 10k
+        internals.setResourceMonitorNetworkUsageThreshold(10 * 1024, 0.001);
+
+        // Skip throttling of unloading.
+        internals.shouldSkipResourceMonitorThrottling = true;
+
+        return true;
+    } else {
+        console.error('ResourceMonitor is not available or cannot modify rules.');
+        return false;
+    }
+}
+
+function rule(filter, action = 'block') {
+    const trigger = { "url-filter": filter };
+
+    return { action: { type: action }, trigger };
+}
+
+async function pause(ms) {
+    return new Promise((resolve) => {
+        setTimeout(() => resolve(), ms);
+    });
+}
+
+async function waitUntilUnload(name) {
+    return new Promise((resolve, reject) => {
+        const iframe = document.querySelector(`iframe[name=${name}]`);
+        if (!iframe)
+            reject();
+
+        const ticket = setInterval(async () => {
```

Comment:
Seems a bit weird to use both `setTimeout` and `setInterval` here. Am I missing something?

It seems like you could just write the straightforward version like:

```
async function waitUntilUnload(name) {
    const iframe = document.querySelector(`iframe[name=${name}]`);
    if (!iframe)
      throw null; // throw in async gets turned in to a promise rejection
    
    while (!iframe.srcdoc) {
      await pause(100);
    }
}
```

Reply:
Right. The script gradually converted to async/await style so that here the mixture was leftover. I'll make it the change.